core-rust/state-manager/src/transaction/executable_logic.rs

```diff
@@ -137,7 +137,7 @@ where
                 "Transaction execution took {}ms, above warning threshold of {}ms ({})",
                 elapsed.as_millis(),
                 self.threshold.as_millis(),
-                self.logged_description
+                (self.logged_description)()
```

Comment:
Hah yes - this was a drive-by-refactor which got out of hand :).

It's not really an issue - just felt weird to have a string created for the purpose of logging if it isn't needed in the main path.

Can revert :)

Reply:
I'd probably stay with a `String` just for sake of simplicity. If it hurts us, then we should probably do what java loggers do (to avoid eagerly rendering a string): pass some `impl Display` there. Maybe an enum like `TimeWarningSubject`?.
Although of course such instance would also need to be created.
And TBH the `Fn` is also a "lightweight struct" (closure) that has to be created and passed, so there is always a cost.
(and we just hope it's more lightweight than rendering a string, but in practice I am sure the difference is not measurable, given how much other computation happens around - that's why I'd go for `String`'s simplicity)